modules/hpe_nimble_volume.py

```diff
@@ -0,0 +1,865 @@
+    parent_vol_name:
+        required: False
+        type: str
+        default: null
+        description:
+        - Name of parent volume
+    snapshot_name:
+        required: False
+        type: str
+        default: null
+        description:
+        - Base snapshot name. This attribute is required together with name and clone when cloning a volume with the create operation
+    volcoll_name:
+        required: False
+        type: str
+        default: null
+        description:
+        - Name of volume collection of which this volume is a member.
+    force:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Forcibly offline, reduce size or change read-only status a volume.
+    force_vvol:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Forcibly move a Virtual Volume.
+    move:
+        required: False
+        type: bool
+        default: False
+        description:
+        - Move a Volume to different Pool.
+
+extends_documentation_fragment: hpenimble
+short_description: "Manage HPE Nimble Storage volume"
+version_added: "2.9"
+'''
+
+EXAMPLES = r'''
+
+      # if state is create , then create a volume if not present. Fails if already present.
+      # if state is "present", then create a volume if not present, But Success if it already exists
+    - name: Create Volume if not present
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: "{{ state | default('present') }}" # fail if exist
+        size: "{{ size }}"
+        limit_iops: "{{ limit_iops }}"
+        limit_mbps: 5000
+        force: false
+        metadata: "{{ metadata }}"
+        description: "{{ description }}"
+        name: "{{ name }}"
+
+    - name: Changing Volume "{{ name }}" to offline state
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: offline
+        name: "{{ name }}"
+
+    - name: Changing Volume "{{ name }}" to online state
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        state: online
+        name: "{{ name }}"
+
+    # Create a clone from the given snapshot name.
+    # if snapshot name is not provided then create a snapshot of source volume
+    # clone task should only run if parent_vol_name is specified.snapshot name is optional
+    - name: Let's create or refresh a clone!
+      hpe_nimble_volume:
+        hostname: "{{ hostname }}"
+        username: "{{ username }}"
+        password: "{{ password }}"
+        name: "{{ name }}" # do we neeed the source vol if snapshot_name is provided?
+        parent_vol_name: "{{ parent_vol_name | mandatory }}"
+        snapshot_name: "{{ snapshot_name | default(None)}}"
```

Comment:
can we use `parent_volume` and `snapshot` here?

Reply:
please reconfirm
https://github.com/hpe-storage/nimble-ansible-modules/pull/1#issuecomment-641045512

Reply:
ok.. will change accordingly